pkg/server/bundle/client/updater.go

```diff
@@ -47,45 +51,63 @@ func NewBundleUpdater(config BundleUpdaterConfig) BundleUpdater {
 }
 
 func (u *bundleUpdater) UpdateBundle(ctx context.Context) (*bundleutil.Bundle, *bundleutil.Bundle, error) {
-	localBundleOrNil, err := fetchBundleIfExists(ctx, u.c.DataStore, u.c.TrustDomain)
+	client, err := u.newClient(ctx)
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to fetch local bundle: %w", err)
+		return nil, nil, err
 	}
 
-	client, err := u.newClient(localBundleOrNil)
+	localFederatedBundleOrNil, err := fetchBundleIfExists(ctx, u.c.DataStore, u.c.TrustDomain)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, fmt.Errorf("failed to fetch local federated bundle: %w", err)
 	}
 
-	endpointBundle, err := client.FetchBundle(ctx)
+	fetchedFederatedBundle, err := client.FetchBundle(ctx)
 	if err != nil {
-		return localBundleOrNil, nil, fmt.Errorf("failed to fetch endpoint bundle: %w", err)
+		return localFederatedBundleOrNil, nil, fmt.Errorf("failed to fetch federated bundle from endpoint: %w", err)
 	}
 
-	if localBundleOrNil != nil && endpointBundle.EqualTo(localBundleOrNil) {
-		return localBundleOrNil, nil, nil
+	if localFederatedBundleOrNil != nil && fetchedFederatedBundle.EqualTo(localFederatedBundleOrNil) {
+		return localFederatedBundleOrNil, nil, nil
 	}
 
-	_, err = u.c.DataStore.SetBundle(ctx, endpointBundle.Proto())
+	_, err = u.c.DataStore.SetBundle(ctx, fetchedFederatedBundle.Proto())
 	if err != nil {
-		return localBundleOrNil, nil, fmt.Errorf("failed to store endpoint bundle: %w", err)
+		return localFederatedBundleOrNil, nil, fmt.Errorf("failed to store fetched federated bundle: %w", err)
 	}
 
-	return localBundleOrNil, endpointBundle, nil
+	return localFederatedBundleOrNil, fetchedFederatedBundle, nil
+}
+
+func (u *bundleUpdater) TrustDomainConfig() TrustDomainConfig {
+	return u.c.TrustDomainConfig
 }
 
-func (u *bundleUpdater) newClient(localBundleOrNil *bundleutil.Bundle) (Client, error) {
+func (u *bundleUpdater) newClient(ctx context.Context) (Client, error) {
 	config := ClientConfig{
-		TrustDomain:     u.c.TrustDomain,
-		EndpointAddress: u.c.EndpointAddress,
+		TrustDomain:      u.c.TrustDomain,
+		EndpointURL:      u.c.EndpointURL,
+		DeprecatedConfig: u.c.DeprecatedConfig,
 	}
-	if !u.c.UseWebPKI {
-		if localBundleOrNil == nil {
-			return nil, errs.New("local bundle not found")
+
+	if spiffeAuth, ok := u.c.EndpointProfile.(HTTPSSPIFFEProfile); ok {
+		trustDomain := spiffeAuth.EndpointSPIFFEID.TrustDomain()
+
+		// This is to preserve behavioral compatibility when using
+		// the deprecated config and will be removed in 1.1.0.
+		if u.c.DeprecatedConfig && trustDomain.IsZero() {
+			trustDomain = u.c.TrustDomain
+		}
+		localEndpointBundle, err := fetchBundleIfExists(ctx, u.c.DataStore, trustDomain)
+		if err != nil {
+			return nil, fmt.Errorf("failed to fetch local bundle: %w", err)
```

Comment:
I have been using the term "local bundle" to refer to the bundle of the trust domain that "this" SPIRE server is serving. Perhaps something like this for this line and the one below addresses it?
```suggestion
			return nil, fmt.Errorf("failed to fetch local copy of bundle for %q: %w", trustDomain, err)
```